automation/jleechanorg_pr_automation/orchestrated_pr_runner.py

```
@@ -0,0 +1,406 @@
+            if state in failing_states:
+                return True
+        return False
+    except Exception as exc:
+        log(f"Error checking PR checks for {repo_full}#{pr_number}: {exc}")
+        return False
+
+
+def ensure_base_clone(repo_full: str) -> Path:
+    # Validate repo_full format defensively
+    if not re.match(r"^[\w.-]+/[\w.-]+$", repo_full):
+        raise ValueError(f"Invalid repository format: {repo_full}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)